### PR TITLE
Configuration setting to use system SSL certificates

### PR DESCRIPTION
**Pull request by tim-goddard-flexcompute**
This is an in-progress patch to support using the OS certificate store (e.g. Windows certificate store) for Flow360 requests, when given a configuration option to change this behavior. Not expected to alter existing default behavior. Further testing is required before taking the "draft" mark off this.

We have a customer who is experiencing SSL-related errors, likely related to an SSL intercepting proxy. The certificates for these are usually distributed to the OS certificate store, but will not be copied to the certipy default bundle. This may resolve such issues, by allowing the OS store which the organisation is managing to be used.

Any advice on testing approach would be welcome. The current unit tests don't make any real HTTP requests, which would be needed to test this functionality. Making those requests would require Internet connectivity where tests are run. In the draft tests I ran, there were issues connecting to the Flow360 API (even public endpoints), because the client is not designed to connect without an access key.

### CODE REVIEW

**Comment by tim-goddard-flexcompute** [comment]
Had a look at that failing matrix test, but it's complaining about a line not altered in this PR, last change dated to 6 months ago. No idea why that would pop out now.

File: flow360/user_config.py [hunk 1]
```diff
@@ -124,5 +131,22 @@ def enable_validation(self):
...
+        -------
+        bool
+            whether to use the OS system certificate store for SSL validation"""
+        self._check_env_profile()
+        self._check_env_usessytemcerts()
+        if self._use_system_certs is not None:
+            return self._use_system_certs
+
+        config_map = self.config.get(self.profile, {})
+        setting = config_map.get("usesystemcerts", False)
```
Comment (maciej-flexcompute) [inline flow360/user_config.py:148]: this one to work requires usesystemcerts being available in config.toml file which is managed by flow360 CLI. You can add this option to the CLI so the user can set it only once: https://github.com/flexcompute/Flow360/blob/develop/flow360/cli/app.py

File: tests/_test_http.py
Comment (maciej-flexcompute) [inline tests/_test_http.py:1]: the `_test_...` are not run automatically in the pipeline. This is why I think this test will have no effect

**Comment by maciej-flexcompute** [comment]
@dominik-flex can you look at this PR? Do we need this or can we close it?

**Comment by dominik-flex** [comment]
@maciej-flexcompute I think we no longer need this.

The solution for requests made with the Python `requests` library is to prepare a custom CA bundle (.pem file) with the Zscaler root certificate in it and set the `REQUESTS_CA_BUNDLE` environment variable to point to that file. It worked for the customer in question.

**Pull request closed without merging**